### PR TITLE
feat: Add parameterization rule custom code UI

### PR DESCRIPTION
**Pull request by e-fisher**
## Description

Add ability to use custom code in parameterization rules. The editor will include read-only function definition to reflect how the snippet will look in the script. In the future, this could be extended to pass arguments to parameterization function (query params, body, headers) and show inline types in the snippet editor. 

## How to Test

- Create a parameterization rule with custom code, verify parameterized value in script and validator
- Verify it's not possible to edit anything other than contents of the wrapper function

## Checklist

- [x] I have performed a self-review of my code.
- [ ] I have added tests for my changes.
- [ ] I have run linter locally (`npm run lint`) and all checks pass.
- [ ] I have run tests locally (`npm test`) and all tests pass.
- [x] I have commented on my code, particularly in hard-to-understand areas.

## Screenshots (if appropriate):

CleanShot 2024-11-11 at 17 11 56

### CODE REVIEW

**Review by Llandy3d — APPROVED** [review]
Reviewing from the functionality standpoint, it works really well! 🚀 🚀 🚀 

It's nice that we can showcase the encasing function in such a native way to the editor 🚀 🙌

File: src/types/constrainedEditor.d.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
+declare module 'constrained-editor-plugin' {
```
Comment (cristianoventura) [inline src/types/constrainedEditor.d.ts:1]: 😮 Are they accepting contributions?

**Review by cristianoventura — APPROVED** [review]
Nicely done! 👏

File: src/types/constrainedEditor.d.ts (hunk 1 above)
Comment (e-fisher) [reply src/types/constrainedEditor.d.ts:1]: The repository doesn't look active lately 😞 But we have already been using this in GCk6 and had types defined there

**Pull request merged**